features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -142,25 +141,26 @@ internal class Base64Serializer private constructor(
         drawableHeight: Int,
         displayMetrics: DisplayMetrics,
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
-        base64SerializerCallback: Base64SerializerCallback?
-    ): Bitmap? {
+        base64SerializerCallback: Base64SerializerCallback
+    ) {
         drawableUtils.createBitmapOfApproxSizeFromDrawable(
             drawable,
             drawableWidth,
             drawableHeight,
-            displayMetrics
-        )?.let { resizedBitmap ->
-            serializeBitmapAsynchronously(
-                drawable,
-                bitmap = resizedBitmap,
-                shouldCacheBitmap = true,
-                imageWireframe,
-                base64SerializerCallback
-            )
-            return resizedBitmap
-        }
-
-        return null
+            displayMetrics,
+            base64SerializerCallback = base64SerializerCallback,
+            bitmapCreationCallback = object : BitmapCreationCallback {
+                override fun onReady(bitmap: Bitmap) {
+                    serializeBitmapAsynchronously(
+                        drawable,
+                        bitmap,
+                        shouldCacheBitmap = true,
+                        imageWireframe,
+                        base64SerializerCallback
```

Comment:
design question: why do we have to pass `base64SerializerCallback` twice? first time as method argument, second time wrapped in `bitmapCreationCallback`?